Simplify shaders (+ one style commit)

JayFoxRox opened the pull request:
We should use v0 to v15 instead of custom vertex attributes. No more special-case handling or forgotten attributes.

I also decided to split the GLSL compilation blocks into a function with better log code.
The new code will return a shader GLuint. I decided against automaticly attaching to a program so we can easily reuse shaders later if we want to do that.

Furthermore I moved the debug routines to the header so we could reuse them in the split nv2a_\* parts.

VP and FFP also use the same constant space so there will be a similar change for constants at some point.
(This will not only simplify the code but also allow us to use uniform blocks for the constants.)

Lastly this also changes one sizeof(x)/sizeof(x[0]) case to use ARRAYSIZE.

Mostly untested [getting SIGSEGV with my GPU driver so I can only test the slow mesa software renderer] - compiles and runs.

JayFoxRox committed:
Move debug routines and macros to nv2a_debug

JayFoxRox committed:
use ARRAYSIZE

JayFoxRox committed:
Use v0 to v15 instead of custom attribs

JayFoxRox committed:
Split GLSL compilation into function

espes commented on hw/xbox/nv2a.h:
```diff
@@ -20,6 +20,84 @@
 #ifndef HW_NV2A_H
 #define HW_NV2A_H
 
+// #define DEBUG_NV2A
```
put this into nv2a_debug.{c,h}

espes commented on hw/xbox/nv2a_psh.c:
```diff
@@ -535,8 +535,8 @@ static QString* psh_convert(struct PixelShader *ps)
     qstring_append(preflight, STRUCT_VERTEX_DATA);
-    qstring_append(preflight, "noperspective in VertexData g_vtx;\n");
     qstring_append(preflight, "#define vtx g_vtx\n");
+    qstring_append(preflight, "noperspective in VertexData vtx;\n");
```
doing the declaration after the alias is less clear

JayFoxRox replied:
probably true, will remove said commit

espes commented on hw/xbox/nv2a_shaders.c:
```diff
@@ -280,10 +290,42 @@ static QString* generate_fixed_function(const ShaderState state,
     return s;
 }
 
+static GLuint create_and_compile_gl_shader(GLenum gl_type,
```
nit: s/create_and_compile_gl_shader/compile_shader/, s/gl_type/gl_shader_type/

JayFoxRox replied:
I agree about gl_shader_type because the GL name was GLuint shaderType.
However, the function name itself should at least show that it's a pure GL helper function IMHO.
I'd accept compile_gl_shader() or create_gl_shader() but I feel compile_shader is not explanatory enough. Ideally all our functions / variable-names should show if they belong to the graphics API. Hopefully that will make it easier to port to something such as vulkan, D3D or software rendering later.

JayFoxRox committed:
Move back NV2A_VERTEX_ATTR_* to nv2a.c

JayFoxRox commented:
Tried to cleanup the PR further.
Moved debug routines to nv2a_debug.c/.h , renamed parameter to gl_shader_type (to match GL shaderType), renamed compile_and_create_gl_shader() to the shorter create_gl_shader() [I still think the _gl_ should be part of the signature because the function is a simple GL helper]. Also removed the vtx patch from earlier.

espes commented:
thanks!
